main.rb

```diff
@@ -0,0 +1,232 @@
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = true if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = true if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = true if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_none?(par = nil)
+    ret_value = true
+    if block_given?
+      i = 0
+      length.times do
+        return false if yield(self[i])
+
+        i += 1
+      end
+    elsif !par
+      i = 0
+      length.times do
+        return false if self[i]
+
+        i += 1
+      end
+    else
+      if par.is_a? Regexp
+        my_each { |x| ret_value = false if x =~ par }
+        return ret_value
+      end
+      if par.is_a? Class
+        my_each { |x| ret_value = false if x.is_a?(par) }
+        return ret_value
+      end
+      unless par.nil?
+        my_each { |x| ret_value = false if x == par }
+      end
+    end
+    ret_value
+  end
+
+  def my_count(par = nil)
+    counter = 0
+    if block_given?
+      i = 0
+      length.times do
+        counter += 1 unless yield(self[i]).nil?
+        i += 1
+      end
+    elsif !par
+      return length
+    else
+      if par.is_a? Regexp
+        my_each { |x| counter += 1 if x =~ par }
+        return counter
+      end
+      if par.is_a? Class
+        my_each { |x| counter += 1 if x.is_a?(par) }
+        return counter
+      end
+      unless par.nil?
+        my_each { |x| counter += 1 if x == par }
+      end
+    end
+    counter
+  end
+
+  def my_map
+    return to_enum unless block_given?
+
+    arr_ret = []
+    i = 0
+    length.times do
+      arr_ret.push(yield(self[i]))
+      i += 1
+    end
+    arr_ret
+  end
+
+  def my_map_proc(proces = nil)
+    i = 0
+    arr = []
+    unless proces.nil?
+      return to_enum unless proces.is_a?(Proc)
+
+      length.times do
+        arr << proces.call(self[i])
+        i += 1
+      end
+      return arr
+    end
+    if block_given?
+      i = 0
+      length.times do
+        arr.push(yield(self[i]))
+        i += 1
+      end
+      arr
+    else
+      to_enum
+    end
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
+    a = self
+    a = Array(self) if is_a? Range
+    op = ['+', '-', '*', '/', '%', '**'].freeze
+    if !arg1.nil? || !arg2.nil?
+      if !arg2.nil?
+        total = arg1
+        i = 0
+        symb = arg2.to_sym if op.include?(arg2.to_s)
+      elsif !arg1.nil?
+        total = first
+        i = 1
+        symb = arg1.to_sym if op.include?(arg1.to_s)
+      end
+      a.length.times do
+        total = total.send(symb, a[i]) if i <= a.length
+        i += 1
+      end
+      return total
+    end
+    if block_given?
+      total = first
+      i = 1
+      (a.length - 1).times do
+        total = yield(total, a[i]) if i < a.length
+        i += 1
+      end
+    end
+    total
+  end
+end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving a TypeError
![image](https://user-images.githubusercontent.com/52209067/77706654-9f798c80-6f88-11ea-9c74-aa4f8b25ad9d.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77706680-b15b2f80-6f88-11ea-9b34-344460d54aac.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving a TypeError
![image](https://user-images.githubusercontent.com/52209067/77707166-f469d280-6f89-11ea-9e71-e8255504f3bc.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77707198-0e0b1a00-6f8a-11ea-8913-54853c63ce26.png)